README.md

```diff
@@ -1,277 +1,899 @@
+enforce that *both* Alice's and Bob's compositions are properly executed when their requests are
+batched together? This is not an issue of standards; this is an issue of whether or not the required
+composition logic can even fit in a transaction where other dApp logic is being executed. The only
+way for the composition to be trustless is if smart contracts enforced it. To put it concretely, in
+Alice's composition, the swap logic and the options logic must both be executed in the same
+transaction where the composition logic is executed. Given the very small amount of execution units
+available to scripts in a transaction, is there even room for this extra logic? The composition
+logic requires two things: it must support *arbitrary* compositions and it must be extremely cheap
+since the logic of the actual dApps being composed must also be executed in the same transaction.
+*P2P protocols do not have the same limitations since Alice personally creates and signs her own
+transaction.* The above two requirements are only requirements when middle-men are creating and
+executing transactions on the behalf of users. Even if some composition logic could fit in the
+transaction, would it actually support arbitrary composition (as opposed to only a few priviledged
+dApps being supported)? Even if the answer to this question is yes, that extra logic being executed
+means less space for other dApps to compose in the transaction. Therefore, protocols that require
+batchers will never be able to offer the the same level of trustless composition as fully P2P
+protocols. The only way to actually match the level of composition of P2P protocols is for the
+batcher based composition to *not be trustless* (without the composition logic, more dApps can fit
+in a transaction).
+- No delegation control or voting control while using the DEX - DEXs try to issue governance tokens
+to get around this but fair distribution of the governance tokens is rarely accomplished (it is a
+very hard problem). Even if the governance tokens were distributed fairly, they are largely just for
+show since the DEX creators can still choose to go against the vote. There is no trustless
+connection between the governance tokens and what actually happens with the stake/dApp. Finally,
+even if there was a trustless connection, there is no way to actually have all users express their
+own stake preferences when the stake is decided democratically. It is almost inevitable that the
+voting outcome will go against the wishes of some users. *This is in direct violation of the
+security assumptions of PoS.* No matter how you look at it, LP based dApps are an existenstial
+problem for PoS blockchains. As LP based dApps become more adopted, the underlying PoS blockchain
+will only become more centralized. LPs can never be used as the foundation for a trustless and
+decentralized PoS blockchain economy.
+
```

Comment:
Maybe add:
A whale that contributes 51% of the liquidity pool assets will get to decide the voting direction.

Reply:
This seems difficult to concisely add since it has a lot of implicit concepts. For example, a whale that controls 51% of staked ada can decide the direction of the blockchain. Why is it okay when talking about Cardano in general but not okay when talking about the DApp? The problem is in how likely it is to acquire the 51% but now I would need to explain the difference. I am honestly not sure how to best articulate this difference. I was hoping my sentence "fair distribution of governance tokens is rarely accomplished..." was enough to allude to control by a whale. Do you really think it is not enough?

Reply:
What you have already I think is OK.  I just wasn't sure if people really get the significance in general.

A whale can actually leverage his voting power by participating in these liquidity pools where majority decides using a governance token.  For example, lets say I have 500M Ada and there are a couple of liquidity pools each with 200M Ada in them.  I can now contribute an extra 201M Ada to each of these liquidity pools so that each pool now has 401M Ada but because I have more than 50% of each pool then I get to choose the voting direction.  I now have control over 802M Ada in voting power plus the extra 98M Ada I still have in my wallet.

There are other scenarios where I could leverage my voting power even more.  For example, I can watch the voting of a particular liquidity pool and then contribute just enough tokens to get the vote over 50% in the direction I wanted, which may require much less tokens than 50% depending on how others in the pool are voting.

Imagine I used this technique to control how Ada was staked to particular pools in my malicious "coalition":  I could provide enough Ada to various liquidity pools and then decide how this Ada was staked and I get the extra Ada from the other smaller liquidity providers in addition to my own.  My coalition of pools thereby gets more Ada staked with it and therefore more control over the block production.

This pooling of assets in liquidity pools is a very bad idea because it hands more power to whales that possess enough capacity to shift the balance.  I don't think most people realise that liquidity pools with governance tokens allow whales to actually increase their power.

Your design fixes this huge threat to control of Cardano's block chain.

Reply:
> I just wasn't sure if people really get the significance in general.

I think this is a scenario where actions speak louder than words. Speculators simply don't care and developers are incentivized not to care since liquidity pools are "what's hot" and where the money is. My design has literally no way for me to profit as the creator which also hurts the chances of developers coming around. Not having an insider token also makes it very difficult to get funded. "It is difficult to get a man to understand something when his salary depends upon his not understanding it." I think (hope) those that are open to listen should be easily swayed by what has already been said.

TBH I think the situation is actually self-correcting. All of the governance issues of liquidity pools are true, but the more immediate concern is that they are economically unstable and will likely collapse within the next several years. Yield farming is just hyper-inflation waiting to happen. Most LPs say their yield farming is temporary but it fundamentally cannot be temporary. How will liquidity be incentivized without it? Transaction fees are *not* paid in the token which means there is no refilling pot like with Cardano's reserves (ideally). Where will the new tokens required to pay liquidity providers come from? These tokens are economically useless as well so the idea that the DApp can provide a service to raise a reserve pot is also extremely unlikely. The only option left is to mint more. 

I think the only reason LPs haven't collapsed yet is because users are staking the yield tokens to earn more yield tokens. Once users decide to "cash out" en masse, like during the beginning of a bear market, the weaknesses will be exposed and I think a lot of users will get hurt. The beginning of a bear market is also when the risk of impermanent loss will be highest which further drives users to sell the yield tokens in order to recoup their losses. I don't see how this doesn't result in the value of the tokens very quickly going to zero. Without any incentive to take on the risk of impermanent loss, users will run for the exits. It would be a run on liquidity pools. I wouldn't be surprised if this scenario resulted in a lot of lawsuits and criminal charges against the developers.

I don't see how the distributed design doesn't win by default.

Reply:
Yep.  I have never participated in any of this liquidity pools - yield farming craziness for exactly the reasons you have articulated.  I think they are worse than Ponzis because aside from hurting the direct users, they can also cause collateral damage to other parts of DeFi through re-hypothecation and other dapps building upon or using governance tokens as collateral.  Furthermore, they might  even hurt the proper functioning of the base protocol due to the control over how their pooled Ada is staked.  They are bad design using dreamed up additional tokens to create Ponzi-like artificial demand.